src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -4942,78 +4942,78 @@ MergeTreeData::DataPartsVector MergeTreeData::Transaction::commit(MergeTreeData:
             }
         }
 
-        NOEXCEPT_SCOPE;
+        NOEXCEPT_SCOPE({
```

Comment:
It's much more cumbersome. But maybe saved exception is worth it...

Reply:
>It's much more cumbersome

Indeed. But this makes the scope explicit, and for noexcept I think it is better to have explicit scope, and it should not be such huge like here. Also even if it should be such huge it can be moved into a separate function.